### PR TITLE
Example: Apply bloom to individual object

### PR DESCRIPTION
**Pull request by Temdog007**
This [comment](https://discourse.threejs.org/t/whats-the-easiest-recommended-way-to-create-glow-layers/4897/5) was inspiration.

### CODE REVIEW

**Comment by Temdog007** [comment]
For testing: https://raw.githack.com/Temdog007/three.js/example/objectBloom/examples/webgl_postprocessing_unreal_bloom_selective.html

**Comment by mrdoob** [comment]
I don't think I would do it this way... The masking produces a strange output.

Instead, I would render the bloom scene with the non-bloomed cubes in black. Then apply the resulting render on top of the normal scene.

File: examples/webgl_postprocessing_unreal_bloom_selective.html [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+			renderer.setSize( window.innerWidth, window.innerHeight );
+			renderer.toneMapping = THREE.ReinhardToneMapping;
+			document.body.appendChild( renderer.domElement );
+
+			scene = new THREE.Scene();
+
+			camera = new THREE.PerspectiveCamera( 40, window.innerWidth / window.innerHeight, 1, 200 );
+			camera.position.set( 0, 0, 30 );
+			camera.lookAt( 0, 0, 0 );
+			camera.updateProjectionMatrix();
```
Comment (WestLangley) [inline examples/webgl_postprocessing_unreal_bloom_selective.html]: You do not not need to update the projection matrix.

File: examples/webgl_postprocessing_unreal_bloom_selective.html [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+			}
+
+			function updateCamera() {
+
+				var distance = params.cameraDistance;
+				var angle = params.cameraAngle;
+				camera.position.x = Math.cos( angle * THREE.Math.DEG2RAD ) * distance;
+				camera.position.z = Math.sin( angle * THREE.Math.DEG2RAD ) * distance;
+				camera.lookAt( 0, 0, 0 );
+				camera.updateProjectionMatrix();
```
Comment (WestLangley) [inline examples/webgl_postprocessing_unreal_bloom_selective.html]: You do not not need to update the projection matrix.

File: examples/webgl_postprocessing_unreal_bloom_selective.html [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+				var columns = params.columns;
+				var rows = params.rows;
+				var size = params.size;
+				scene.children.length = 0;
+				for ( var x = - columns * size / 2; x < columns * size / 2; x += size ) {
+
+					for ( var y = - rows * size / 2; y < rows * size / 2; y += size ) {
+
+						var box = new THREE.Mesh( new THREE.BoxBufferGeometry( size, size, size ), new THREE.MeshBasicMaterial( {
+							color: Math.floor( Math.random() * 0xffffff )
```
Comment (WestLangley) [inline examples/webgl_postprocessing_unreal_bloom_selective.html]: try making the boxes darker.

File: examples/webgl_postprocessing_unreal_bloom_selective.html [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+						} ) );
+						box.position.set( x, y, 0 );
```
Comment (WestLangley) [inline examples/webgl_postprocessing_unreal_bloom_selective.html]: for more interest,
```js
box.position.set( x, y, Math.random() );
```

File: examples/webgl_postprocessing_unreal_bloom_selective.html [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+						scene.add( box );
+						if ( Math.random() < 0.5 ) box.layers.enable( BLOOM_SCENE );
```
Comment (WestLangley) [inline examples/webgl_postprocessing_unreal_bloom_selective.html]: I suggest fewer
```js
if ( Math.random() < 0.125 )
```

File: examples/webgl_postprocessing_unreal_bloom_selective.html [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+
+					}
+
+				}
+
+			}
+
+			function animate() {
```
Comment (WestLangley) [inline examples/webgl_postprocessing_unreal_bloom_selective.html]: You do not need an animation loop for a static scene.

**Comment by WestLangley** [comment]
This is considerably-improved from the previous version. :-)

You are not disposing your objects correctly. Type `renderer.info` into the console to see what is happening.

Consider using `OrbitControls` without an animation loop.

**Comment by mrdoob** [comment]
Thanks!

**Comment by WestLangley** [comment]
Nice work @Temdog007 !

I deleted the temporary live link.

**Comment by mrdoob** [comment]
I ended up simplifying the example quite a bit 😇
https://threejs.org/examples/#webgl_postprocessing_unreal_bloom_selective

**Comment by WestLangley** [comment]
Oh. That was you that did that! Looks great!

**Comment by WestLangley** [comment]
We should revisit this pattern:
```js
bloomComposer.setSize( window.innerWidth * window.devicePixelRatio, window.innerHeight * window.devicePixelRatio );
```
I don't think we are consistent in how we handle DPR in `EffectComposer`.

**Comment by mrdoob** [comment]
Definitely. That was a last minute hack 😕

**Comment by Mugen87** [comment]
I've tried to clean up this issue with #14340

Apart of `webgl_postprocessing_unreal_bloom_selective`, post-processing example do not handle the resize correctly since `window.devicePixelRatio` is not taken into account. The idea of the PR was to handle the pixel ratio automatically in `EffectComposer.setSize()`. This would make `EffectComposer` more consistent since it already uses the drawing buffer's size when creating render targets.

**Comment by WestLangley** [comment]
TBH, I've already spent more time on that issue than I care to, so whatever you want to do about it is fine with me.

I do not care if it is handled automatically, as long as it is correct, and we are consistent in the pattern we recommend users to follow.

**Pull request merged**